src/ig/IGExporter.ts

```diff
@@ -1151,18 +1154,19 @@ export class IGExporter {
   addIgIni(igPath: string): void {
     const inputIniPath = path.join(this.igDataPath, 'ig.ini');
     if (this.config.template != null) {
+      const filePathString = path.join(path.basename(this.igDataPath), 'ig.ini');
       if (existsSync(inputIniPath)) {
-        const filePathString = path.join(path.basename(this.igDataPath), 'ig.ini');
-
         let preferredFileMessage =
           `Since a ${filePathString} file was found, the "template" property in the ${this.configName} ` +
-          `will be ignored and an ig.ini file will not be generated. Remove the ${filePathString} ` +
-          `file to use the "template" property in ${this.configName} and generate an ig.ini file instead.`;
+          'will be ignored and an ig.ini file will not be generated. Support for the "template" property ' +
+          `in ${this.configName} will be removed in a future version of SUSHI. Please remove the "template" ` +
+          `property in ${this.configName} and manage the ig.ini file directly.`;
```

Comment:
Something feels fishy about how these errors and warnings are lining up. What I'm seeing is:
* If I'm using an `input/fsh` folder and I have `template` in my `sushi-config.yaml` and no `ig.ini` file, I get an error telling me not to use `template`, with what would be the contents of the ini file.
* If I'm using an `input/fsh` folder and I have `template` in my `sushi-config.yaml` and I do have `ig.ini`, I get a warning telling me that my `template` property will be ignored.

I feel like the second case should be an error as well. If you disagree, then we will have to change the `package-list` PR, since the situation for `history` is analogous to the one for `template`, but the errors and warnings being logged are not currently analogous.

Reply:
So... I was thinking that if you don't have `ig.ini` already, then the end result would be a project that _won't_ build w/ the publisher.  Clearly an error condition.  If you have an `ig.ini` and we're just not going to overwrite it, then you probably still have a valid IG overall -- so that's more of a warning condition.

Somewhat (but not totally) analogous: If your config has a `menu` property, but you also have an `input/includes/menu.xml` file, we won't overwrite the file and we log a warning.  So it's analogous in that we're warning when you have duplicate sources; it's not analogous in that `menu` property is not deprecated.

Re: template and ig.ini: I could probably be convinced either way.  What do you think?

Reply:
I could also go either way. I'm happy as long as package-list and this match in when it is an error and a warning. My logic was that we are only letting `template` hang around for people who have legacy IGs. If you are making an IG with the new structure, and you specify `template` or `history`, I liked having an error, because if we weren't supporting legacy modes we wouldn't let you have those config properties at all. So I didn't want to let people get away with something they shouldn't in the new structure, just because we need to allow it for the legacy structure.

Reply:
OK.  Let's make it an error then!